guarddog/scanners/scanner.py

```diff
@@ -186,9 +185,23 @@ def scan_local(
             }
         """
 
+        requirement_paths = []
+
         try:
-            with open(path, "r") as f:
-                return self.scan_requirements(f.read(), rules, callback)
+            if os.path.isfile(path):
+                requirement_paths.append(path)
+            elif os.path.isdir(path):
+                requirement_paths.extend(self.find_requirements(path))
+            else:
+                raise Exception(f"unable to find file or directory {path}")
```

Comment:
I think is better to avoid using the generic one
```suggestion
                raise ValueError(f"unable to find file or directory {path}")
```

Reply:
added in 64ad725